packages/tenant-process/src/services/readModelService.ts

```diff
@@ -0,0 +1,162 @@
+import { z } from "zod";
+import { logger, ReadModelRepository } from "pagopa-interop-commons";
+import {
+  agreementState,
+  ErrorTypes,
+  ListResult,
+  Tenant,
+} from "pagopa-interop-models";
+import { AggregationCursor, Document } from "mongodb";
+import { config } from "../utilities/config.js";
+
+const { tenants } = ReadModelRepository.init(config);
+
+/*
+function arrayToFilter<T, F extends object>(
+  array: T[],
+  f: (array: T[]) => F
+): F | undefined {
+  return array.length > 0 ? f(array) : undefined;
+}
+*/
+
+async function getTotalCount(
+  query: AggregationCursor<Document>
+): Promise<number> {
+  const data = await query.toArray();
+  const result = z.array(z.object({ count: z.number() })).safeParse(data);
+
+  if (result.success) {
+    return result.data.length > 0 ? result.data[0].count : 0;
+  }
+
+  logger.error(
+    `Unable to get total count from aggregation pipeline: result ${JSON.stringify(
+      result
+    )} - data ${JSON.stringify(data)} `
+  );
+  throw ErrorTypes.GenericError;
+}
```

Comment:
I was watching other PRs and I saw that the `getTotalCount` function has been moved inside the `ReadModelRepository` class. We could get rid of the re-declaration here and use it from there

Reply:
Updated, thanks